pypeit/scripts/ql_keck_lris.py

```diff
@@ -0,0 +1,475 @@
+"""
+Script for quick-look reductions of Keck MOSFIRE observations.
+
+.. include common links, assuming primary doc root is up one directory
+.. include:: ../include/links.rst
+"""
+
+import os
+import copy
+from glob import glob
+import time
+
+from pkg_resources import resource_filename
+
+from IPython import embed
+
+import numpy as np
+
+from astropy.io import fits
+from astropy.table import Table
+from astropy.stats import sigma_clipped_stats
+
+from pypeit import utils
+from pypeit import pypeit
+from pypeit import par, msgs
+from pypeit import pypeitsetup
+from pypeit import wavecalib
+from pypeit import wavetilts
+from pypeit import spec2dobj
+from pypeit import coadd2d
+from pypeit import specobjs
+from pypeit import slittrace
+from pypeit import reduce
+from pypeit import calibrations
+from pypeit.display import display
+from pypeit.images import buildimage
+from pypeit.spectrographs.util import load_spectrograph
+from pypeit.core.parse import get_dnum, parse_binning
+from pypeit.core.wavecal import wvutils
+from pypeit import sensfunc
+from pypeit.core import flux_calib
+from pypeit.scripts import scriptbase
+
+
+def config_lines(args):
+    # Config the run
+    cfg_lines = ['[rdx]']
+    cfg_lines += ['    spectrograph = {0}'.format('keck_lris_red_mark4')]
+    cfg_lines += ['    redux_path = {0}'.format(args.redux_path)]
+    cfg_lines += ['    scidir = Science_QL']
+    # Calibrations
+    cfg_lines += ['[baseprocess]']
+    cfg_lines += ['    use_pixelflat = False']
+    cfg_lines += ['    use_illumflat = False']
+    cfg_lines += ['    use_biasimage = False'] # TODO fix for LRIS
+
+    cfg_lines += ['[calibrations]']
+    cfg_lines += ['    [[wavelengths]]']
+    cfg_lines += ['        refframe = observed']
+    cfg_lines += ['[scienceframe]']
+    cfg_lines += ['    [[process]]']
+    cfg_lines += ['        spat_flexure_correct = False']
+    if not args.mask_cr:
+        cfg_lines += ['        mask_cr = False']
+    cfg_lines += ['[reduce]']
+    cfg_lines += ['    [[extraction]]']
+    cfg_lines += ['        skip_optimal = True']
+    if args.box_radius is not None:  # Boxcar radius
+        cfg_lines += ['        boxcar_radius = {0}'.format(args.box_radius)]
+    cfg_lines += ['    [[findobj]]']
+    cfg_lines += ['        skip_second_find = True']
+    cfg_lines += ['[flexure]']
+    cfg_lines += ['        spec_method = skip']
+    return cfg_lines
+
+
+def run(files, caliBrate, spectrograph, det, parset, show=False, std_trace=None):
+    """
+    Peform 2d extraction for a set of files at the same unique A-B offset location.
```

Comment:
A-B for LRIS?

Reply:
This has been refactored. Review in upcoming PR.